code/lightna/magento-os-frontend/js/common/AddToCart.js

```diff
@@ -1,71 +1,65 @@
-import { Cookie } from '../lib/Cookie';
 import { UserInput } from '../lib/UserInput';
 import request from '../lib/HttpClient';
-import { $, $$ } from "../lib/utils/dom";
+import { $ } from "../lib/utils/dom";
 
 export class AddToCart {
-    components = [];
     addToCartUrl = '/checkout/cart/add';
     classes = {
         loading: 'loading',
         disabled: 'btn-disabled',
     };
 
     constructor() {
-        this.components = $$('.cjs-add-to-cart');
+        this.component = '.cjs-add-to-cart';
+        this.trigger = '[data-action="add-to-cart"]';
         this.bindEvents();
     }
 
     bindEvents() {
-        this.components.forEach((component) => {
-            const actionTrigger = $('[data-action="add-to-cart"]', component);
-            actionTrigger.addEventListener('click', () => {
-                this.addProduct(component, actionTrigger);
-                this.toggleAnimation(actionTrigger, true);
-            });
+        $('body').addEventListener('click', (event) => {
+            const trigger = this.getAddToCartTrigger(event.target);
+            if (trigger) {
+                const component = trigger.closest(this.component);
+                this.onAddProductStart(trigger);
+                this.addProduct(component)
+                    .then(this.onAddProductComplete.bind(this, trigger));
```

Comment:
I'n not sure we have a complete here, complete triggers even if error was thrown, but it's not triggered in our case. Dunno if we need true complete event.
maybe beforeAddProduct/afterAddProduct?

Reply:
🫡